lib/puppet/util/execution.rb

```diff
@@ -336,6 +338,11 @@ def self.execute_posix(command, options, stdin, stdout, stderr)
       # turning "/tmp/foo;\r\n /bin/echo" into ["/tmp/foo;\r\n", " /bin/echo"]
       command = [command].flatten
       Process.setsid
+      if options[:priority]
+        if Process.uid == 0
```

Comment:
We typically use `Puppet.features.root?` for this check.

Reply:
fixed, thanks